### DISCUSSION TITLE
[Announcement] Homework 3 is out

### DISCUSSION THREAD

**Discussion by yeonhee-ryou** [prosyslab-classroom/cs348-information-security#459] [Announcements]
Hi all,
this is an announcement about homework 3.

Homework 3 is a programming assignment requiring you to implement encryption, decryption and crack algorithms of RSA.

Here is the GitHub classroom link for this homework:
(https://classroom.github.com/a/MHfVVR71)

## Auto grading
Your implementation will be graded using GradeScope.
Please check [GradeScope guideline](https://github.com/prosyslab-classroom/cs348-information-security/blob/main/TOOL.md#gradescope).

Memory and time limits are important in this assignment.
For each test case, score will be given as follow:
```ocaml
let score = match (execute test_case) with
  | Success(execution_time) -> 60.0 - execution_time
  | Fail -> 0.0
  | Timeout -> 0.0
```

## Due date
Considering the midterm period, the due date for homework 3 is 2 weeks from today.

Due: 04. 25 (Tue.) 23:59:59
Late Due: 04. 27 (Thu.) 23:59:59 (late submission penalty).

**Comment by lummsklr** [comment]
I got a score but the autograder marks each function red. Is it okay?
![image](https://user-images.githubusercontent.com/78457167/231431694-1696345c-617e-4a8f-8931-3a74b630166e.png)

**Comment by yeonhee-ryou** [comment]
pass/fail marks have the same meaning as before: #236

**Comment by lummsklr** [comment]
Thank you

**Comment by bongjunj** [comment]
What is the size of machine-word on the grader machine?
The size of `int` depends on machine-word size in OCaml.
And it can be useful to know this to implement the HW.

**Comment by KihongHeo** [comment]
maybe https://v2.ocaml.org/api/Sys.html#VALint_size ?

**Comment by yeonhee-ryou** [comment]
> > Note that we limit the maximum of the input upperbound to 2^16 (=65536).
> 
> Doe it means `./rsa keygen 65536` and `dec`, `enc` should pass? Or the maximum value of the upperbound is 65535?

The maximum input value of the `ub` parameter of `keygen`.
(+ You may infer the upper bounds of other inputs of `dec` and `enc`)

**Comment by gayeun4195** [comment]
Is there any general rule for generating e and d in keygen? p, q is described as biggest two prime numbers same with or smaller than ub. However, I think there can exist various e and d even if p and q is fixed.

**Comment by jinho-choi123** [comment]
I think we have to choose the smallest e to get the correct answer
